### PR TITLE
Document e2e test command options.

### PR DESCRIPTION
**Pull request by sainthkh**
## Description

Document e2e test command options to make it work correctly on all platforms.

## How has this been tested?

Manually tested.

## Screenshots 

N/A

## Types of changes

New feature for developers.

## Checklist:
- [N/A] My code is tested.
- [N/A] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [N/A] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: docs/contributors/testing-overview.md [hunk 1]
```diff
@@ -387,14 +387,26 @@ or interactively
...
+Sometimes it's useful to observe the browser while running tests. Then, use this command:
+
+```bash
+npm run test-e2e:open-browser
+```
+
+You can control the speed of execution by changing environment variables like below:
 
 ```bash
 PUPPETEER_HEADLESS=false PUPPETEER_SLOWMO=80 npm run test-e2e:watch
```
Comment (gziolo) [inline docs/contributors/testing-overview.md:399]: There is an option that does the same:
```suggestion
npm run test-e2e:watch -- --puppeteer-interactive --puppeteer-slowmo=200
```

https://github.com/WordPress/gutenberg/blob/33a821ff00755e7a535e30b69bb0588941c3b908/packages/scripts/scripts/test-e2e.js#L59-L62

File: docs/contributors/testing-overview.md [hunk 2]
```diff
@@ -387,14 +387,26 @@ or interactively
...
 You can additionally have the devtools automatically open for interactive debugging in the browser:
 
+```bash
+npm run test-e2e:devtools
+```
+
+It's the short version of the long command below:
+
 ```bash
 PUPPETEER_HEADLESS=false PUPPETEER_DEVTOOLS=true npm run test-e2e:watch
```
Comment (gziolo) [inline docs/contributors/testing-overview.md:411]:
```suggestion
npm run test-e2e:watch -- --puppeteer-devtools
```

This one also has a nicer version:
https://github.com/WordPress/gutenberg/blob/33a821ff00755e7a535e30b69bb0588941c3b908/packages/scripts/scripts/test-e2e.js#L64-L67

File: package.json [hunk 3]
```diff
@@ -223,6 +223,8 @@
 		"test": "npm run lint && npm run test-unit",
 		"test-e2e": "wp-scripts test-e2e --config packages/e2e-tests/jest.config.js",
 		"test-e2e:debug": "wp-scripts --inspect-brk test-e2e --config packages/e2e-tests/jest.config.js --puppeteer-devtools",
+		"test-e2e:open-browser": "cross-env PUPPETEER_HEADLESS=false PUPPETEER_SLOWMO=80 wp-scripts test-e2e --config packages/e2e-tests/jest.config.js",
+		"test-e2e:devtools": "cross-env PUPPETEER_HEADLESS=false PUPPETEER_SLOWMO=80 PUPPETEER_DEVTOOLS=true wp-scripts test-e2e --config packages/e2e-tests/jest.config.js",
```
Comment (gziolo) [inline package.json:227]: You can also enable DevTools with a keyboard shortcut while running the tests in the browser so I don't think it deserves it's own script here.

File: package.json [hunk 4]
```diff
@@ -223,6 +223,8 @@
 		"test": "npm run lint && npm run test-unit",
 		"test-e2e": "wp-scripts test-e2e --config packages/e2e-tests/jest.config.js",
 		"test-e2e:debug": "wp-scripts --inspect-brk test-e2e --config packages/e2e-tests/jest.config.js --puppeteer-devtools",
+		"test-e2e:open-browser": "cross-env PUPPETEER_HEADLESS=false PUPPETEER_SLOWMO=80 wp-scripts test-e2e --config packages/e2e-tests/jest.config.js",
```
Comment (gziolo) [inline package.json:226]: With updates added to the documentation, do you think it's still valuable to have this script included here? It's not that much to type `npm run test-e2e:watch -- --puppeteer-interactive`.

Comment (sainthkh) [reply package.json:226]: I agree. I don't think we need them any more. I didn't know that the options existed.

Comment (gziolo) [reply package.json:226]: I don't know why it wasn't documented :(

**Comment by sainthkh** [comment]
Changed the title of this PR to `Document e2e test command options.` and edited the content.

File: package.json (hunk 4 above)
Comment (sainthkh) [reply package.json]: It's how things work in software development. Computer wants everything but we sometimes miss one or two. That's why we work together to find those missing pieces.

File: docs/contributors/testing-overview.md [hunk 5]
```diff
@@ -387,22 +387,28 @@ or interactively
...
 ```bash
-PUPPETEER_HEADLESS=false PUPPETEER_DEVTOOLS=true npm run test-e2e:watch
+npm run test-e2e:watch -- --puppeteer-devtools
 ```
 
 If you're using a different setup, you can provide the base URL, username and password like this:
 
 ```bash
-WP_BASE_URL=http://localhost:8888 WP_USERNAME=admin WP_PASSWORD=password npm run test-e2e
+npm run test-e2e -- --wordpress-base-url=http://localhost:8888 --wordpress-username=admin --wordpress-password=password
```
Comment (gziolo) [inline docs/contributors/testing-overview.md:411]: I completely forgot about that, thank you for bringing it up as well :)

Comment (gziolo) [reply docs/contributors/testing-overview.md:411]: So it was added in #14717 but it didn't touch this file.

**Review by gziolo — APPROVED** [review]
Awesome, thank you for the initiative 👍

**Pull request merged**